tests/unit/test_get_cosine_decay_spline.py

```diff
@@ -0,0 +1,76 @@
+"""
+Tests of `get_cosine_decay_harmonised_spline`
+"""
+
+from __future__ import annotations
+
+import numpy as np
+import pytest
+
+from gradient_aware_harmonisation import get_cosine_decay_harmonised_spline
+from gradient_aware_harmonisation.timeseries import Timeseries
+from gradient_aware_harmonisation.utils import add_constant_to_spline
+
+scipy = pytest.importorskip("scipy")
+
+
+@pytest.mark.parametrize("harmonisation_time", [3.0, 3.5, 4.0])
+@pytest.mark.parametrize("convergence_time", [6.0, 9.0, 10.0])
+def test_get_cosine_decay(harmonisation_time, convergence_time):
```

Comment:
When I look at this, this seems to be testing the same thing as `tests/integration/test_harmonise_splines_integration.py` so we could delete this (although it doesn't hurt to have it so ultimately up to you).